spec/controllers/bookings_controller_spec.rb

```diff
@@ -0,0 +1,21 @@
+require "rails_helper"
+
+RSpec.describe BookingsController, type: :request do
+  let(:event) { Event.create(start_time: 1.hour.from_now, duration: 90, address: "Louvain La Neuve", capacity: 9) }
+  let(:user) { User.create(email: "user@example.com", password: "password", admin: true) }
+
+  describe "#create" do
+    before do
+      sign_in user
+    end
+    context "with valid parameters" do
+      it "creates a new booking" do
+        post event_bookings_path(event), params: { booking: { user_id: user.id } }
+        expect(response).to have_http_status(:redirect)
+        expect(response).to redirect_to(root_path)
+        expect(Booking.last.event).to eq(event)
+        expect(Booking.last.user).to eq(user)
+      end
+    end
```

Comment:
The booking's user is taken from the current_user inside your controller, not from a parameter so you don't need the params.

Can you one test to verify that the cancel action works too please ?
If you not be canceled at first then after patching with your cancel_booking route it should then be canceled